glusterd2/volume/volume-utils.go

```diff
@@ -198,4 +198,9 @@ func CreateVolumeInfoResp(v *Volinfo) *api.VolumeInfo {
 		Metadata:  v.Metadata,
 		SnapList:  v.SnapList,
 	}
+	if len(volinfo.SnapList) == 0 {
+		volinfo.SnapList = []string{}
```

Comment:
Well, this isn't neat although it fixes the problem.

What do you think of using `omitempty` on `snap-list` instead ?

Reply:
if we want to use ```omitempty``` for this. a volume may have empty metadata and options as well.same should be used  for ```options``` and ```metadata```.
let me know your thought on this.

Reply:
Okay.

Reply:
Wait, a better place to do this is at `newVolinfo` in `glusterd2/commands/volumes/volume-create-txn.go`. Let me know if that's an acceptable approach.

Reply:
I think this is the best place to do. as ```CreateVolumeInfoResp``` will be called by other functions for preparing volume response ex:- snapshot, VolumeExpand etc

Reply:
@prashanthpai's suggestion is to initialize when Volume is created, so that you need not worry formatting while sending response.